tests/cpp/test_sharding.cpp

```diff
@@ -407,18 +407,20 @@ TEST_F(ShardingTest, ShardedInnerReshape) {
 TEST_F(ShardingTest, ShardedReshapeWithIndependentSplit) {
   auto fusion = std::make_unique<Fusion>();
   FusionGuard fg(fusion.get());
-  DeviceMesh mesh({0, 1});
-  int64_t d = mesh.size();
+  DeviceMesh mesh = DeviceMesh::createForShape({2, 3});
```

Comment:
```suggestion
  auto mesh = DeviceMesh::createForShape({2, 3});
```
Type::createXxx is enough to indicate it returns Type or a variant of it. So using `auto` here is fine and preferred for conciseness.